### PR TITLE
support as-raw requests

### PR DESCRIPTION
**Pull request by ShanniLi**
r @willsalz @truncs

a simple start. :)

### CODE REVIEW

**Comment by truncs** [comment]
@ShanniLi Raw requests are already supported, the only thing that is required is use the call method with a RawRequest and it would return a RawResponse. Infact I am benchmarking right now with a RawRequest and RawResponses.

File: tchannel-core/src/main/java/com/uber/tchannel/schemes/RawSerializer.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    private static final Gson GSON = new Gson();
+
+    @Override
+    public String decodeEndpoint(ByteBuf arg1) {
+        String endpoint = arg1.toString(CharsetUtil.UTF_8);
+        arg1.release();
+        return endpoint;
+    }
+
+//    @Override
```
Comment (willsalz) [inline tchannel-core/src/main/java/com/uber/tchannel/schemes/RawSerializer.java]: nit: don't commit commented code

File: tchannel-core/src/main/java/com/uber/tchannel/schemes/RawSerializer.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        arg3.release();
+        return body;
+    }
+
+    @Override
+    public ByteBuf encodeEndpoint(String method) {
+        return Unpooled.wrappedBuffer(method.getBytes());
+    }
+
+//    @Override
```
Comment (willsalz) [inline tchannel-core/src/main/java/com/uber/tchannel/schemes/RawSerializer.java]: same here.

File: tchannel-core/src/main/java/com/uber/tchannel/schemes/RawSerializer.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+//    public <T> T decodeHeaders(ByteBuf arg2) {
+//        String header = arg2.toString(CharsetUtil.UTF_8);
+//        arg2.release();
+//        return (T)header;
+//    }
+
+    @Override
+    public Map<String, String> decodeHeaders(ByteBuf arg2) {
```
Comment (willsalz) [inline tchannel-core/src/main/java/com/uber/tchannel/schemes/RawSerializer.java]: This seems incorrect; why are you decoding raw args at all?

File: tchannel-core/src/main/java/com/uber/tchannel/schemes/RawSerializer.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        String headerJSON = arg2.toString(CharsetUtil.UTF_8);
+        arg2.release();
+        Map<String, String> headers = GSON.fromJson(headerJSON, HEADER_TYPE);
+        return (headers != null) ? headers : new HashMap<String, String>();
+    }
+
+    @Override
+    public <T> T decodeBody(ByteBuf arg3, Class<T> bodyType) {
```
Comment (willsalz) [inline tchannel-core/src/main/java/com/uber/tchannel/schemes/RawSerializer.java]: Again, the Raw Arg Scheme makes no assumptions about the body; so why is this being decoded as a string? And subsequently cast to a `T`?

To make the downstream consumer's life easy, we should be returning `byte[]`, then `releasing()` the reference counted buffer to make sure that the client isn't responsible for that.

**Comment by ShanniLi** [comment]
@truncs, the reason for this RP is to provide a consistent interface for as-raw like others. as-raw is actually heavily used ...
@willsalz very good points! I will put this PR to shelf first. I will need to understand more about as-thrift to make a better version of this PR.

ref: #84

**Pull request closed without merging**